client/src/app/components/deployment-selection/deployment-selection.component.ts

```diff
@@ -16,13 +17,26 @@ export class DeploymentSelectionComponent {
   queryClient = injectQueryClient();
 
   sourceRef = input.required<string>();
+  repositoryId = input.required<number>();
+
+  permissionsQuery = injectQuery(() => ({
+    ...getRepoPermissionsOptions({
+      path: { repoId: this.repositoryId() }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
nitpick: this is pretty dirty. I would recommend to change the server code as described in my comment below